harmony_netcdf_to_zarr/mosaic_utilities.py

```diff
@@ -384,13 +494,25 @@ def get_grid_values(input_values: np.ndarray,
         the input array to the maximum. The grid spacing will be the supplied
         resolution.
 
+        The scale factor is used to ensure the `np.linspace` output does not
+        include recurring decimals. For example, the GPM/IMERG longitude grid
+        output dimension previously included -179.8499999999, instead of
+        -179.85.
```

Comment:
This is a pain. The general effect can be reproduced with `numpy`:

```
In [1]: import numpy as np
In [2]: input_vals = np.linspace(-179.5, 179.5, 3600, dtype=np.float32)
In [3]: input_vals
Out[3]:
array([-179.5    , -179.40025, -179.3005 , ...,  179.3005 ,  179.40025,
        179.5    ], dtype=float32)

```
The rounding below seems to have fixed the issue for the longitude dimension in GPM/IMERG, and not adversely affected all other test dimensions (MERRA-2: latitude, longitude, time and GPM/IMERG: latitude and time).